### ISSUE TITLE
Prophet's fit() method cannot handle dates prior to 1728-01-01

### ISSUE THREAD

**Issue by rayjennings3rd** [facebook/prophet#1867]
I have a dataset that goes back to 1700 up to the future. The fit() method will raise an exception because pandas cannot handle subtracting the current date from a date < 1728-01-01

OutOfBoundsDatetime: Result is too large for pandas.Timedelta. Convert inputs to datetime.datetime with 'Timestamp.to_pydatetime()' before subtracting.

Prophet method "setup_data()" will automatically convert your "ds" column to panda datetime - regardless of what value you pass in as your "ds" column.

I tried to start doing a conversion, just to see how far I could get to using pything datetime.datetime - instead of pandas time.

The exact line the exception is thrown is:

self.t_scale = df['ds'].max() - self.start

**Comment by bletham** [comment]
Yeah this is happening in pandas, here is a simple repro:
```Python
import pandas as pd
df = pd.DataFrame({'ds': ['1700-01-01', '2022-01-01']})
df['ds'] = pd.to_datetime(df['ds'])
df['ds'].max() - df['ds'].min()  # OverflowError: Python int too large to convert to C long
```
When you take a difference in pandas datetimes, it creates a pd.Timedelta. The default unit for a pd.Timedelta is nanoseconds; so here it is storing the number of nanoseconds between 1700-01-01 and 2022-01-01 which can't fit in a 32-bit integer.

I'm not sure how easy this will be to fix. The particular line you highlight would need to instead construct a pd.Timedelta with a different unit (e.g. microseconds):
https://github.com/facebook/prophet/blob/cd8a24eddd6bebb9888f2e0ae2388f209047b02d/python/prophet/forecaster.py#L354
And it would have to be done here too:
https://github.com/facebook/prophet/blob/cd8a24eddd6bebb9888f2e0ae2388f209047b02d/python/prophet/forecaster.py#L325
I'm sure that's possible I'm just not sure how involved it will end up being. There may be other uses of pd.datetime that would have to be fixed too; I'm not sure.

Unfortunately there isn't an obvious workaround to me since everything internally is pretty tightly tied to datetimes. Well, I guess the workaround is to truncate the history; I would be curious what the application here is and if it is reasonable to use only more recent data.

**Comment by rayjennings3rd** [comment]
Yes, I tried to convert things just as a curiosity but got to a point where the exception turned into "python datetime not being able to fit into C type long."

This application was for sunspots. I think the tradeoff of being able to go back to 1700 vs not having nanosecond granularity in a time series would be worth it.

**Comment by bletham** [comment]
Interesting. Yes I agree that nanosecond granularity is not necessary for any type of time series that Prophet would be likely to be useful for, and changing that to microseconds would give 1000x the range with no loss whatsoever.
